resolver/blocking_resolver.go

```diff
@@ -515,12 +527,12 @@ func (r *BlockingResolver) groupsToCheckForClient(request *model.Request) []stri
 
 func (r *BlockingResolver) matches(groupsToCheck []string, m lists.Matcher,
 	domain string,
-) (group []string) {
+) (matches map[string]string) {
 	if len(groupsToCheck) > 0 {
 		return m.Match(domain, groupsToCheck)
 	}
 
-	return []string{}
+	return map[string]string{}
```

Comment:
Let's make it nil while we change it since we don't need to allocate an empty map.
```suggestion
	return nil
```